pkg/restic/repository_manager.go

```diff
@@ -141,72 +142,55 @@ func (rm *repositoryManager) NewRestorer(ctx context.Context, restore *arkv1api.
 		},
 	)
 
-	r := newRestorer(ctx, rm, informer, rm.repoLister)
+	r := newRestorer(ctx, rm, rm.repoEnsurer, informer)
 
 	go informer.Run(ctx.Done())
-	if !cache.WaitForCacheSync(ctx.Done(), informer.HasSynced) {
+	if !cache.WaitForCacheSync(ctx.Done(), informer.HasSynced, rm.repoInformerSynced) {
 		return nil, errors.New("timed out waiting for cache to sync")
 	}
 
 	return r, nil
 }
 
-func (rm *repositoryManager) InitRepo(name string) error {
-	repo, err := getRepo(rm.repoLister, rm.namespace, name)
-	if err != nil {
-		return err
-	}
-
+func (rm *repositoryManager) InitRepo(name, identifier string) error {
 	rm.repoLocker.LockExclusive(name)
 	defer rm.repoLocker.UnlockExclusive(name)
 
-	return rm.exec(InitCommand(repo.Spec.ResticIdentifier))
+	return rm.exec(InitCommand(identifier))
 }
 
-func (rm *repositoryManager) CheckRepo(name string) error {
-	repo, err := getRepo(rm.repoLister, rm.namespace, name)
-	if err != nil {
-		return err
-	}
-
+func (rm *repositoryManager) CheckRepo(name, identifier string) error {
 	rm.repoLocker.LockExclusive(name)
 	defer rm.repoLocker.UnlockExclusive(name)
 
-	cmd := CheckCommand(repo.Spec.ResticIdentifier)
-
-	return rm.exec(cmd)
+	return rm.exec(CheckCommand(identifier))
 }
 
-func (rm *repositoryManager) PruneRepo(name string) error {
-	repo, err := getReadyRepo(rm.repoLister, rm.namespace, name)
-	if err != nil {
-		return err
-	}
-
+func (rm *repositoryManager) PruneRepo(name, identifier string) error {
 	rm.repoLocker.LockExclusive(name)
 	defer rm.repoLocker.UnlockExclusive(name)
 
-	cmd := PruneCommand(repo.Spec.ResticIdentifier)
-
-	return rm.exec(cmd)
+	return rm.exec(PruneCommand(identifier))
 }
 
-func (rm *repositoryManager) Forget(snapshot SnapshotIdentifier) error {
-	repo, err := getReadyRepo(rm.repoLister, rm.namespace, snapshot.Repo)
+func (rm *repositoryManager) Forget(ctx context.Context, snapshot SnapshotIdentifier) error {
+	if !cache.WaitForCacheSync(ctx.Done(), rm.repoInformerSynced) {
+		return errors.New("timed out waiting for cache to sync")
+	}
+
+	repo, err := rm.repoEnsurer.EnsureRepo(ctx, rm.namespace, snapshot.Repo)
```

Comment:
This has the potential to create the restic repo. We shouldn't be doing that as part of deletions/GC.

Reply:
same question re: spinning up a new cluster, then having GC process backups that were synced in from an existing bucket.

Reply:
Yup, my bad